worker/uniter/pebblenotices.go

```diff
@@ -131,11 +131,59 @@ func (n *pebbleNoticer) run(containerName string) (err error) {
 	}
 }
 
-func (n *pebbleNoticer) processNotice(containerName string, notice *client.Notice) error {
+func (n *pebbleNoticer) processNotice(containerName string, notice *client.Notice, pebbleClient PebbleClient) error {
 	var eventType container.WorkloadEventType
+	var event container.WorkloadEvent
 	switch notice.Type {
 	case client.CustomNotice:
 		eventType = container.CustomNoticeEvent
+		event = container.WorkloadEvent{
+			Type:         eventType,
+			WorkloadName: containerName,
+			NoticeID:     notice.ID,
+			NoticeType:   string(notice.Type),
+			NoticeKey:    notice.Key,
+		}
+	case client.ChangeUpdateNotice:
+		data := notice.LastData
+		// Since the charm is triggering most Pebble changes, there is little
+		// value in feeding events based on those changes back to the charm.
+		// As such, we do not send events for all change-updated notices (see
+		// OP045 for more background). However, perform-check and recover-check
+		// change kinds reflect changes in the workload state, so are useful to
+		// charms.
+		if data["kind"] != "perform-check" && data["kind"] != "recover-check" {
```

Comment:
You could pull out `kind := data["kind"]` above as a named variable, seeing you're using it 6x here.